server/datastore/mysql/queries.go

```diff
@@ -256,9 +297,18 @@ func (ds *Datastore) NewQuery(
 	return query, nil
 }
 
+func (ds *Datastore) updateQueryLabels(ctx context.Context, query *fleet.Query) error {
+	return ds.updateQueryLabelsInTx(ctx, query, nil)
+}
+
 // updates the LabelsIncludeAny for a query, using the string value of
 // the label. Labels IDs are populated
-func (ds *Datastore) updateQueryLabels(ctx context.Context, query *fleet.Query) error {
+func (ds *Datastore) updateQueryLabelsInTx(ctx context.Context, query *fleet.Query, txToUse *sqlx.Tx) error {
+	var (
+		tx  *sqlx.Tx
+		err error
+	)
+
```

Comment:
Split this into two methods. `updateQueryLabelsInTx` is the workhorse, and will either use the transaction given to it, or will start its own.  `updateQueryLabels` wraps `updateQueryLabelsInTx` and calls it without providing a transaction.

The spec API that GitOps calls ends up calling `ApplyQueries`, which starts its own transaction, so it calls `updateQueryLabelsInTx` directly using that transaction.  This way if any problems occur applying labels to queries, the whole transaction can be rolled back.